### PR TITLE
VACMS 0000: Some other random PHPStan fixes

### PR DESCRIPTION
**Pull request by ndouglas**
## Description

Featuring dependencies on modules we no longer use, removing unused services, adding backslashes before namespaces, etc.  I'm trying not to change any actual executed code, just improve the quality of the comments and type hints, etc.

## QA steps

As user _uid_ with _user_role_
1. Do this
1. Then that
1. Then validate Acceptance Criteria from issue
- [ ] This
- [ ] That
- [ ] The other thing

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `Core Application Team`
- [ ] `Product Support Team`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change? 
- [ ] Yes, and it's written in issue ____ and queued for publication. 
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written 
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [ ] No announcement is needed for this code change. 
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

File: docroot/modules/custom/va_gov_govdelivery/src/Service/ProcessStatusBulletin.php [hunk 1]
```diff
@@ -367,6 +367,7 @@ private function getSendType(NodeInterface $node, $latest_situation_update) : st
         // This is a new situation update that needs to be sent.
         return self::SEND_TYPE_SITUATION;
       }
+      return '';
```
Comment (ndouglas) [inline docroot/modules/custom/va_gov_govdelivery/src/Service/ProcessStatusBulletin.php:370]: This actually changes executed code and I was a little iffy about it.  

But it is only called from this code AFAICT:

```php
    $send_type = $this->getSendType($node, $situation_update);
    if (empty($send_type)) {
      return;
    }
```

`empty()` ["is essentially the concise equivalent to"](https://www.php.net/manual/en/function.empty.php) `!isset($var) || $var == false` and `"" == false`.  I felt a pang of doubt and double-checked myself, lest I wreck myself.

File: docroot/modules/custom/va_gov_backend/src/Plugin/field_group/FieldGroupFormatter/Tooltip.php [hunk 2]
```diff
@@ -200,7 +156,7 @@ public static function defaultContextSettings($context) {
     $defaults = [
       'open' => FALSE,
       'required_fields' => $context == 'form',
-    ] + parent::defaultSettings($context);
+    ] + parent::defaultSettings();
```
Comment (ndouglas) [inline docroot/modules/custom/va_gov_backend/src/Plugin/field_group/FieldGroupFormatter/Tooltip.php:159]: This method doesn't accept a parameter, so I think it was a typo or a mistake.

File: docroot/modules/custom/va_gov_backend/src/Plugin/field_group/FieldGroupFormatter/Tooltip.php [hunk 3]
```diff
@@ -21,48 +18,7 @@
...
-   * @param string $label
-   *   The formatter label.
-   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityManager
-   *   The entity instance.
-   */
-  public function __construct($plugin_id, $plugin_definition, \stdClass $group, array $settings, $label, EntityTypeManagerInterface $entityManager) {
-    parent::__construct($plugin_id, $plugin_definition, $group, $settings, $label);
-    $this->entityManager = $entityManager;
-  }
+class Tooltip extends HtmlElement {
```
Comment (ndouglas) [inline docroot/modules/custom/va_gov_backend/src/Plugin/field_group/FieldGroupFormatter/Tooltip.php:21]: None of this was necessary, probably left over from a more complicated implementation.

File: docroot/modules/custom/va_gov_backend/src/Plugin/field_group/FieldGroupFormatter/DetailsWithImage.php [hunk 4]
```diff
@@ -21,12 +21,13 @@
...
+
   /**
-   * The entity manager service.
+   * The module handler service.
    *
-   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   * @var \Drupal\Core\Extension\ModuleHandlerInterface
    */
-  protected $entityManager;
+  protected $moduleHandler;
```
Comment (ndouglas) [inline docroot/modules/custom/va_gov_backend/src/Plugin/field_group/FieldGroupFormatter/DetailsWithImage.php:30]: The entity type manager service was never used at any point and I think was just copied and pasted boilerplate code.  The module handler service was used, however, so I injected it 😎

File: docroot/modules/custom/va_gov_backend/src/Plugin/Validation/Constraint/BenefitsSelectionLimitValidator.php [hunk 5]
```diff
@@ -14,6 +14,7 @@ class BenefitsSelectionLimitValidator extends ConstraintValidator {
    * {@inheritdoc}
    */
   public function validate($item, Constraint $constraint) {
+    /** @var \Drupal\va_gov_backend\Plugin\Validation\Constraint\BenefitsSelectionLimit $constraint */
```
Comment (ndouglas) [inline docroot/modules/custom/va_gov_backend/src/Plugin/Validation/Constraint/BenefitsSelectionLimitValidator.php:17]: I don't really like the Symfony validation and constraint pattern, and I _really_ don't like it as it's used in Drupal.  It feels kinda clumsy and is not really amenable to static analysis, so I ended up having a pattern of changes to make to these files.

File: docroot/modules/custom/va_gov_workflow_assignments/src/Plugin/Block/EntityMetaDisplay.php [hunk 6]
```diff
@@ -83,7 +83,7 @@ public function __construct(
     RouteMatchInterface $route_match,
     EntityTypeManagerInterface $entity_type_manager,
     ExclusionTypesInterface $exclusionTypes,
-    VaGovUrl $vaGovUrl,
+    VaGovUrlInterface $vaGovUrl,
```
Comment (ndouglas) [inline docroot/modules/custom/va_gov_workflow_assignments/src/Plugin/Block/EntityMetaDisplay.php:86]: in ur codebase, interfacing with ur vagov

File: tests/behat/Drupal/FeatureContext.php [hunk 7]
```diff
@@ -17,7 +17,6 @@ class FeatureContext extends RawDrupalContext implements SnippetAcceptingContext
   use \Traits\FieldTrait;
   use \Traits\UserEntityTrait;
   use \Traits\ContentTrait;
-  use \Traits\GroupTrait;
```
Comment (ndouglas) [inline tests/behat/Drupal/FeatureContext.php:20]: This appears to depend on the `group` module, which is not in our codebase.  I wasn't able to determine if it has ever actually been enabled 😕   I nuked it, so let's see if tests pass 🍿

File: docroot/modules/custom/va_gov_workflow_assignments/src/Plugin/Block/EntityMetaDisplay.php (hunk 6 above)
Comment (acrollet) [reply docroot/modules/custom/va_gov_workflow_assignments/src/Plugin/Block/EntityMetaDisplay.php:86]: 🤓

File: docroot/modules/custom/va_gov_bulk/src/Service/ModerationActionsInterface.php [hunk 8]
```diff
@@ -5,22 +5,22 @@
 use Drupal\node\NodeInterface;
 
 /**
- * Interface ModerationActionsInterface.
+ * Encapsulates logic for bulk moderation actions.
```
Comment (acrollet) [inline docroot/modules/custom/va_gov_bulk/src/Service/ModerationActionsInterface.php:8]: ❤️

**Review by acrollet — APPROVED** [review]
shipit

**Pull request merged**